src/tests/system/tests/test_pysss_nss_idmap.py

```diff
@@ -0,0 +1,345 @@
+"""
+SSSD Python Client identity Lookups
+
+:requirement: Python binding of libsss_nss_idmap.so
+"""
+
+from __future__ import annotations
+
+import ast
+
+import pytest
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.roles.samba import Samba
+from sssd_test_framework.topology import KnownTopology, KnownTopologyGroup
+
+
+def get_sid_name(provider: GenericProvider):
+    if isinstance(provider, Samba):
+        return "objectSid"
+    elif isinstance(provider, AD):
+        return "SID"
+    elif isinstance(provider, IPA):
+        return "ipaNTSecurityIdentifier"
+    return None
+
+
+def user_setup(provider: GenericProvider, client: Client):
```

Comment:
The next two functions, I think we should extend the user and group classes in the framework. Other tests can use this feature.